src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,63 @@
+package duke.task;
+
+import duke.DukeException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline.
+ */
+public class Deadline extends Task {
+
+    protected LocalDate byDate;
+    protected LocalTime byTime;
+
+    /**
+     * Constructs a Deadline with the specified description and date and/or time.
+     *
+     * @param description The specified description of the deadline.
+     * @param by The specified date and/or time of the deadline.
+     * @throws DukeException If the date or time are wrongly formatted.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        try {
+            String[] dateAndTime = by.split(" ");
+            this.byDate = LocalDate.parse(dateAndTime[0]);
+            //time is optional
+            if (dateAndTime.length == 2) {
+                this.byTime = LocalTime.parse(dateAndTime[1]);
+            } else {
+                this.byTime = null;
+            }
+        } catch (DateTimeParseException e) {
+            throw new DukeException("OOPS!!! Format is yyyy-mm-dd HH:mm. Time is optional.");
```

Comment:
You can have more specific DukeException for dedicated exception type. E.g. InvalidInputFormatException which extends  your DukeException. It will help the class which catches the error to handle more effectively (e.g. reacting different ways to different exceptions)